nixos/modules/services/mail/postal.md

```diff
@@ -0,0 +1,108 @@
+# Postal {#module-services-postal}
+
+[Postal](https://github.com/postalserver/postal) is an open source mail
+delivery platform for incoming & outgoing e-mail.
+
+## Basic usage {#module-services-postal-basic-usage}
+
+Start by setting up your [domain and DNS](https://docs.postalserver.io/getting-started/dns-configuration).
+
+Except for secrets, which should be stored in [`services.postal.environmentFile`](#opt-services.postal.environmentFile),
+[Postal configuration](https://docs.postalserver.io/getting-started/configuration)
+can be defined in [`services.postal.config`](#opt-services.postal.enable).
+
+**This example is unsafe, don't store secrets in the nix store.**
+
+```nix
+let
+  # generate a rails secret key: `openssl rand -hex 64`
+  # smtp password will need to be set later, once you have generated credentials in Postal’s UI
+  unsafeEnvironmentFile = pkgs.writeText "env-file" ''
+    RAILS_SECRET_KEY=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
+    SMTP_PASSWORD=
+  '';
+
+  # generate a signing key: `openssl genrsa 4096`
+  unsafeSigningKey = pkgs.writeText "signingKey" ''
+    -----BEGIN PRIVATE KEY-----
+    MIIBVQIBADANBgkqhkiG9w0BAQEFAASCAT8wggE7AgEAAkEAtssGlCncaEQfOP7y
+    50ZuTTacnwv5Twz4dBDE3L/JeIE0HOvcX+Na0h4DPbn276mciYhkS4RFpQawgkig
+    j2NuZQIDAQABAkBWYsSNKOtc6zTGPtaUrhharUB/ea0syrhwQayHlqukIr6k7uBC
+    isR3ozdDOHCJVN1iAjWvIPk+m/Nk+RmPKLKBAiEA5bBKyCn1NGo7XqXMvZ3RpY+k
+    mYFuHaAC2/IfVj6Mf50CIQDLu4C7/TydV+s/iRWPy6p7Xb2kGHzpCimrfgoGkcFD
+    aQIhALHQuRQc52ecljm/wbFJ7HNvsM3mFYl5xrzfxMLPyZVBAiBLaPHzo36GNv7K
+    m7Exco997mq9jJrfn3VhFtwbJmRE0QIhAM7fzud2xue48DlTvvD7TsrjyGGw+Tha
+    Vn45rvYNYAyq
+    -----END PRIVATE KEY-----
+  '';
+
+in {
+  networking.firewall.allowedTCPPorts = [ 443 465 ];
```

Comment:
Can we maybe add an `openFirewall` option to the module?

Reply:
Should it only open the SMTP (`smtp_server.default_port`) port? For the HTTP port (`web_server.default_port`) I would not open it because the service should be behind a proxy, maybe I can add an option to configure a nginx vhost but I'm not sure if it's a good practice. Also there is healthchecks ports (for the smtp server and each worker), it could be a separate option but not sure either for them :)

Reply:
Quite a good question, let's skip this and document which ports one should open.

Reply:
After consideration I think smtp is the primary service to expose, the other one would be the web UI, which depends on the proxy and is easy to setup aside, I went ahead with this and added a note about the different ports in the `openFirewall` option.